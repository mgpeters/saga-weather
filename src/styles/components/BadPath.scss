.weather-display--bad-path {
  overflow: hidden;
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: 'Zilla Slab', serif;
  font-size: 2vw;
}
