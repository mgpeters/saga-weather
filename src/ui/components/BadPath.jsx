/**
 * ************************************
 *
 * @module  BadPath.jsx
 * @author  mpeters
 * @date    10/05/2020
 * @description BadPath component for improper
 * URL path
 *
 * ************************************
 */

import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../styles/components/BadPath.scss';

const BadPath = () => {
  return (
    <div className="weather-display--bad-path">
      <h1>BadPathname - Please Enter Another</h1>
    </div>
  );
};

BadPath.propTypes = {
  dailyIndex: PropTypes.object,
  formatTime: PropTypes.func,
  weatherIcon: PropTypes.func,
  toggleModal: PropTypes.func,
};

export default BadPath;
