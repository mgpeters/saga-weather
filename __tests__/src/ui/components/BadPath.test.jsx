import React from 'react';
import { shallow } from 'enzyme';

import BadPath from '../../../../src/ui/components/BadPath.jsx';

describe('BadPath component', () => {
  const wrapper = shallow(<BadPath />);

  it('matches previous snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a 1 <div>', () => {
    const div = wrapper.find('div');
    expect(div).toHaveLength(1);
  });
  it('should have a 1 <h1>', () => {
    const h1 = wrapper.find('h1');
    expect(h1).toHaveLength(1);
    expect(h1.text()).toBe('BadPathname - Please Enter Another');
  });
});
