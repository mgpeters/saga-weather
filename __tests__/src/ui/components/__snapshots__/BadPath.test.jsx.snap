// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadPath component matches previous snapshot 1`] = `ShallowWrapper {}`;
